src/OpenFOAM/CMakeLists.txt

```diff
@@ -609,4 +609,14 @@ target_link_libraries(OpenFOAM PUBLIC Pstream)
 target_include_directories(OpenFOAM PUBLIC
   .
 )
+
+find_package(ZLIB REQUIRED)
+if(ZLIB_FOUND)
+  add_definitions(-DHAVE_LIBZ)
+endif()
+
+# link openfoam with zlib
+target_link_libraries(OpenFOAM PUBLIC ${ZLIB_LIBRARIES})
```

Comment:
We have a target_link_libraries call here: https://github.com/openfoamdotcom/openfoam/blob/d29f00600d42321444580c3d49d167045e01bce4/src/OpenFOAM/CMakeLists.txt#L608

Reply:
And we should be able to use `ZLIB::ZLIB` as the link library so that we don't need the target include directories.